```
10:14:10 FAIL: testFlushDenormalsCPU (__main__.DenormalTest)
10:14:10 ----------------------------------------------------------------------
10:14:10 Traceback (most recent call last):
10:14:10   File "C:/tf_jenkins/home/workspace/tensorflow-pr-win-cmake-py/tensorflow/python/kernel_tests/denormal_test.py", line 50, in testFlushDenormalsCPU
10:14:10     self._flushDenormalsTest(use_gpu=False, dtypes=(np.float32, np.float64))
10:14:10   File "C:/tf_jenkins/home/workspace/tensorflow-pr-win-cmake-py/tensorflow/python/kernel_tests/denormal_test.py", line 44, in _flushDenormalsTest
10:14:10     self.assertAllEqual(flush.eval(), np.zeros(shape))
10:14:10   File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\framework\test_util.py", line 535, in assertAllEqual
10:14:10     np.testing.assert_array_equal(a, b)
10:14:10   File "C:\Program Files\Anaconda3\lib\site-packages\numpy\testing\utils.py", line 871, in assert_array_equal
10:14:10     verbose=verbose, header='Arrays are not equal')
10:14:10   File "C:\Program Files\Anaconda3\lib\site-packages\numpy\testing\utils.py", line 796, in assert_array_compare
10:14:10     raise AssertionError(msg)
10:14:10 AssertionError: 
10:14:10 Arrays are not equal
10:14:10 
10:14:10 (mismatch 100.0%)
10:14:10  x: array(1.1754946310819804e-39, dtype=float32)
10:14:10  y: array(0.0)
10:14:10 
10:14:10 ======================================================================
10:14:10 FAIL: testFlushDenormalsGPU (__main__.DenormalTest)
10:14:10 ----------------------------------------------------------------------
10:14:10 Traceback (most recent call last):
10:14:10   File "C:/tf_jenkins/home/workspace/tensorflow-pr-win-cmake-py/tensorflow/python/kernel_tests/denormal_test.py", line 54, in testFlushDenormalsGPU
10:14:10     self._flushDenormalsTest(use_gpu=True, dtypes=(np.float32,))
10:14:10   File "C:/tf_jenkins/home/workspace/tensorflow-pr-win-cmake-py/tensorflow/python/kernel_tests/denormal_test.py", line 44, in _flushDenormalsTest
10:14:10     self.assertAllEqual(flush.eval(), np.zeros(shape))
10:14:10   File "C:\Program Files\Anaconda3\lib\site-packages\tensorflow\python\framework\test_util.py", line 535, in assertAllEqual
10:14:10     np.testing.assert_array_equal(a, b)
10:14:10   File "C:\Program Files\Anaconda3\lib\site-packages\numpy\testing\utils.py", line 871, in assert_array_equal
10:14:10     verbose=verbose, header='Arrays are not equal')
10:14:10   File "C:\Program Files\Anaconda3\lib\site-packages\numpy\testing\utils.py", line 796, in assert_array_compare
10:14:10     raise AssertionError(msg)
10:14:10 AssertionError: 
10:14:10 Arrays are not equal
10:14:10 
10:14:10 (mismatch 100.0%)
10:14:10  x: array(1.1754946310819804e-39, dtype=float32)
10:14:10  y: array(0.0)
```

I suppose we could use `assertAllClose` instead of `assertAllEqual` unless it should be exactly zero.

--- a/tensorflow/contrib/cmake/CMakeLists.txt
+++ b/tensorflow/contrib/cmake/CMakeLists.txt
@@ -58,6 +58,7 @@ if(WIN32)
   add_definitions(-DNOMINMAX -D_WIN32_WINNT=0x0A00 -DLANG_CXX11 -DCOMPILER_MSVC -D__VERSION__=\"MSVC\")
   add_definitions(-DWIN32 -DOS_WIN -D_MBCS -DWIN64 -DWIN32_LEAN_AND_MEAN -DNOGDI -DPLATFORM_WINDOWS)
   add_definitions(-DTENSORFLOW_USE_EIGEN_THREADPOOL -DEIGEN_HAS_C99_MATH -D_ITERATOR_DEBUG_LEVEL=0)
+  add_definitions(-DEIGEN_VECTORIZE_SSE3)  # Needed to suppress denormals without __SSE3__ in MSVC
   add_definitions(-DNDEBUG /O2)  # Equivalent of -c opt in Bazel.
   add_definitions(/bigobj /nologo /EHsc /GF /FC /MP /Gm-)
   # Suppress warnings to reduce build log size.

--- a/tensorflow/core/platform/denormal.cc
+++ b/tensorflow/core/platform/denormal.cc
@@ -14,7 +14,9 @@ limitations under the License.
 ==============================================================================*/
 
 #include "tensorflow/core/platform/denormal.h"
-#ifdef __SSE3__
+#include "third_party/eigen3/Eigen/Core"
+// Check EIGEN_VECTORIZE_SSE3 since Windows doesn't define __SSE3__ properly
+#ifdef EIGEN_VECTORIZE_SSE3
 #include <pmmintrin.h>
 #endif
 
@@ -24,7 +26,7 @@ namespace port {
 ScopedFlushDenormal::ScopedFlushDenormal() {
 // For now, we flush denormals only on SSE 3.  Other architectures such as ARM
 // can be added as needed.
-#ifdef __SSE3__
+#ifdef EIGEN_VECTORIZE_SSE3
   // Save existing flags
   flush_zero_mode_ = _MM_GET_FLUSH_ZERO_MODE() == _MM_FLUSH_ZERO_ON;
   denormals_zero_mode_ = _MM_GET_DENORMALS_ZERO_MODE() == _MM_DENORMALS_ZERO_ON;
@@ -38,7 +40,7 @@ ScopedFlushDenormal::ScopedFlushDenormal() {
 }
 
 ScopedFlushDenormal::~ScopedFlushDenormal() {
-#ifdef __SSE3__
+#ifdef EIGEN_VECTORIZE_SSE3
   // Restore flags
   _MM_SET_FLUSH_ZERO_MODE(flush_zero_mode_ ? _MM_FLUSH_ZERO_ON
                                            : _MM_FLUSH_ZERO_OFF);

